core/gdxsv/gdxsv_network.cpp

```diff
@@ -26,12 +26,68 @@ bool TcpClient::Connect(const char *host, int port) {
     memcpy(&addr.sin_addr, host_entry->h_addr_list[0], host_entry->h_length);
 #endif
     addr.sin_port = htons(port);
-    set_recv_timeout(new_sock, 5000);
+    
+    fd_set setW, setE;
+    struct timeval timeout = {5,0};
+    int res;
+    
+    auto set_blocking_mode = [](const int &socket, bool is_blocking) {
+#ifdef _WIN32
+        u_long flags = is_blocking ? 0 : 1;
+        ioctlsocket(socket, FIONBIO, &flags);
+#else
+        const int flags = fcntl(socket, F_GETFL, 0);
+        fcntl(socket, F_SETFL, is_blocking ? flags ^ O_NONBLOCK : flags | O_NONBLOCK);
+#endif
+    };
+    
+    set_blocking_mode(new_sock, false);
+    
+    ::connect(new_sock, (const sockaddr *) &addr, sizeof(addr));
 
-    if (::connect(new_sock, (const sockaddr *) &addr, sizeof(addr)) != 0) {
+#ifdef _WIN32
+    if(get_last_error() != WSAEWOULDBLOCK){
+#else
+    if(get_last_error() != EINPROGRESS){
+#endif
         WARN_LOG(COMMON, "Connect fail 2 %d", get_last_error());
         return false;
+    } else {
+        do {    
+            FD_ZERO(&setW);
+            FD_SET(new_sock, &setW);
+            FD_ZERO(&setE);
+            FD_SET(new_sock, &setE);
+            
+            res = select(new_sock+1, NULL, &setW, &setE, &timeout);
+            if (res < 0 && errno != EINTR) {
+                WARN_LOG(COMMON, "Connect fail 3 %d", get_last_error());
+                return false;
+            } else if (res > 0) {
+                
+                if (FD_ISSET(new_sock, &setE)){
+#ifdef _WIN32
+                    char error;
+#else
+                    int error;
+#endif
+                    socklen_t l = sizeof(int);
```

Comment:
maybe invalid, should be 1 on windows

Reply:
Since it can write `WSAEACCES` (10013) into `error` the length should larger than 1 byte?
Updated to use `int error` on Windows also, and type cast to `(char*)` when calling `getsockopt`